hazelcast/serialization/service.py

```diff
@@ -159,3 +319,194 @@ def register_class_definition(self, class_definition, factories, check_error):
+
+    def serializer_for(self, obj):
+        """Searches for a serializer for the provided object
+
+        Serializers will be  searched in this order;
+
+        - NULL serializer
+        - Default serializers, like primitives, arrays, string and some default types
+        - Custom registered types by user
+        - Global serializer if registered by user
+        - pickle serialization as a fallback
+
+        Args:
+            obj: Input object
+
+        Returns:
+            The serializer
+        """
+        # 1-NULL serializer
+        if obj is None:
+            return self._null_serializer
+
+        obj_type = type(obj)
+
+        # 2-Default serializers, DataSerializable, Portable, Compact,
+        # primitives, arrays, String, UUID and some helper types(BigInteger etc)
+        serializer = self.lookup_default_serializer(obj_type, obj)
+
+        # 3-Custom registered types by user
+        if serializer is None:
+            serializer = self.lookup_custom_serializer(obj_type)
+
+        # 5-Global serializer if registered by user
+        if serializer is None:
+            serializer = self.lookup_global_serializer(obj_type)
+
+        # 4 Internal serializer
+        if serializer is None:
+            serializer = self.lookup_python_serializer(obj_type)
+
+        if serializer is None:
+            raise HazelcastSerializationError(
+                "There is no suitable serializer for:" + str(obj_type)
+            )
+        return serializer
+
+    def lookup_default_serializer(self, obj_type, obj):
+        if isinstance(obj, IdentifiedDataSerializable):
+            return self._data_serializer
+
+        if isinstance(obj, Portable):
+            return self._portable_serializer
+
+        if isinstance(obj, str):
+            return self.serializer_by_type_id(CONSTANT_TYPE_STRING)
+
+        if obj_type in self._compact_types:
+            return self._compact_stream_serializer
+
+        # LOCATE NUMERIC TYPES
+        if obj_type is int:
+            type_id = self._int_type_id
+            if type_id is None:
+                # VAR size
+                if MIN_BYTE <= obj <= MAX_BYTE:
+                    type_id = CONSTANT_TYPE_BYTE
+                elif MIN_SHORT <= obj <= MAX_SHORT:
+                    type_id = CONSTANT_TYPE_SHORT
+                elif MIN_INT <= obj <= MAX_INT:
+                    type_id = CONSTANT_TYPE_INTEGER
+                elif MIN_LONG <= obj <= MAX_LONG:
+                    type_id = CONSTANT_TYPE_LONG
+                else:
+                    type_id = JAVA_DEFAULT_TYPE_BIG_INTEGER
+
+            return self.serializer_by_type_id(type_id)
+
+        return self._constant_type_dict.get(obj_type, None)
+
+    def lookup_custom_serializer(self, obj_type):
+        serializer = self._type_dict.get(obj_type, None)
+        if serializer is not None:
+            return serializer
+        for super_type in obj_type.__subclasses__():
```

Comment:
Yeah maybe. But as I said, this code is not introduced in this PR and I don't want to change it here. Can you create an issue for it, so that someone can look after it later?